src/SAML2/XML/samlp/LogoutRequest.php

```diff
@@ -315,32 +223,27 @@ public function setSessionIndex(string $sessionIndex = null): void
      */
     public function toXML(?DOMElement $parent = null): DOMElement
     {
-        if ($this->encryptedNameId === null && $this->nameId === null) {
-            throw new Exception('Cannot convert LogoutRequest to XML without a NameID set.');
-        }
+        Assert::null($parent);
 
-        $root = parent::toXML();
+        $e = parent::toXML($parent);
 
         if ($this->notOnOrAfter !== null) {
-            $root->setAttribute('NotOnOrAfter', gmdate('Y-m-d\TH:i:s\Z', $this->notOnOrAfter));
+            $e->setAttribute('NotOnOrAfter', gmdate('Y-m-d\TH:i:s\Z', $this->notOnOrAfter));
         }
 
         if ($this->reason !== null) {
-            $root->setAttribute('Reason', $this->reason);
+            $e->setAttribute('Reason', $this->reason);
         }
 
-        if ($this->encryptedNameId === null) {
-            $this->nameId->toXML($root);
-        } else {
-            $eid = $root->ownerDocument->createElementNS(Constants::NS_SAML, 'saml:EncryptedID');
-            $root->appendChild($eid);
-            $eid->appendChild($root->ownerDocument->importNode($this->encryptedNameId, true));
-        }
+        $this->identifier->toXML($e);
```

Comment:
I have to say I love how this ended up being so simple, no matter if the identifier was encrypted or not 😄